src/txn/txn.c

```diff
@@ -1896,6 +1901,9 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
     if (locked)
         __wt_readunlock(session, &txn_global->visibility_rwlock);
 
+    /* Leave the commit generation after all the updates are processed. */
+    __wt_session_gen_leave(session, WT_GEN_TXN_COMMIT);
```

Comment:
I think the comment should be leave the generation after we have processed the updates and release the snapshot. Releasing the snapshot is important here.

Reply:
Thanks @quchenhao. I have updated the comment in [here](93de7559b6c2e02194d62ee02f56fa756610acdf)